### DISCUSSION TITLE
Feature Request: withColumns in Spark API

### DISCUSSION THREAD

**Discussion by MarcoGorelli** [duckdb/duckdb#15363] [Ideas]
The [DuckDB Spark API](https://duckdb.org/docs/api/python/spark_api.html) currently has:
- select
- withColumn

Is [withColumns](https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.DataFrame.withColumns.html) also in-scope?

> Why do you want this feature

To get good DuckDB support in [Narwhals](https://github.com/narwhals-dev/narwhals)

**Comment by binste** [comment]
Added in #14347 :)

**Reply by MarcoGorelli** [reply]
Amazing thanks!
